src/layers/image/rotation.cpp

```diff
@@ -66,30 +66,62 @@ void rotation_layer<TensorDataType, Layout, Device>::fp_compute() {
           const auto& angle_rad = angle * Pi / degree;
 
           // Get center pixel for rotation
-          const El::Int row_center = input_width/2;
-          const El::Int col_center = input_height/2;
+          const El::Int col_center = input_width/2;
+          const El::Int row_center = input_height/2;
 
           // Rotate point relative to input pixel centers
-          const auto& rotated_row =  (output_row - row_center) * cos(angle_rad) - (output_col - col_center) * sin(angle_rad) + row_center;
           const auto& rotated_col = (output_row - row_center) * sin(angle_rad) + (output_col - col_center) * cos(angle_rad) + col_center;
+          const auto& rotated_row =  (output_row - row_center) * cos(angle_rad) - (output_col - col_center) * sin(angle_rad) + row_center;
 
           // Find input pixels near rotation point
-          auto input_col = static_cast<El::Int>(std::floor(rotated_col - half));
-          auto input_row = static_cast<El::Int>(std::floor(rotated_row - half));
+          const auto input_col = static_cast<El::Int>(std::floor(rotated_col));
+          const auto input_row = static_cast<El::Int>(std::floor(rotated_row));
 
           // Input and output pixels
           auto& pixel_output = local_output(channel * input_height * input_width
                                                 + output_row * input_width
                                                 + output_col,
                                                 sample);
 
-	  if((input_row >= 0 && input_row < input_height) && (input_col >= 0 && input_col < input_width)){
-          	auto& pixel_input = local_input(channel * input_height * input_width
-                                       	     	  + input_row * input_width
-                                           	  + input_col,
-                                                  sample);
+	  if((input_col >= 0 && input_col < input_width) && (input_row >= 0 && input_row < input_height)){
```

Comment:
The behavior is slightly incorrect on the left and bottom edges. If `input_col==input_width-1`, then `input_col0==input_col1==input_width-1`. The effect is a constant region on the edge.
```suggestion
	  if((input_col >= 0 && input_col < input_width-1) && (input_row >= 0 && input_row < input_height-1)){
```